### PR TITLE
librfu_intr

### PR DESCRIPTION
**Pull request by PikalaxALT**
This PR depends on pret/agbcc#30.
This PR includes #250.

### CODE REVIEW

File: asm/librfu_intr.s [hunk 1]
```diff
@@ -4,694 +4,3 @@
 	.syntax unified
```
Comment (jiangzhengwenjz) [inline asm/librfu_intr.s]: Remove this file

File: src/librfu_intr.c [hunk 2]
```diff
@@ -0,0 +1,403 @@
+#include "global.h"
```
Comment (jiangzhengwenjz) [inline src/librfu_intr.c]: use constants from `include/gba/io_reg.h`for  io register literals in this file

File: ld_script.txt [hunk 3]
```diff
@@ -322,6 +322,7 @@ SECTIONS {
         src/agb_flash_mx.o(.text);
         src/agb_flash_le.o(.text);
         src/librfu_stwi.o(.text);
+        src/librfu_intr.o(.text);
         asm/librfu_intr.o(.text);
```
Comment (jiangzhengwenjz) [inline ld_script.txt]: remove the line

File: src/librfu_intr.c [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+    {
+        if (
+            gSTWIStatus->ackActiveCommand == (0x80 | ID_MS_CHANGE_REQ)
+         || gSTWIStatus->ackActiveCommand == (0x80 | ID_DATA_TX_AND_CHANGE_REQ)
+         || gSTWIStatus->ackActiveCommand == (0x80 | ID_UNK35_REQ)
+         || gSTWIStatus->ackActiveCommand == (0x80 | ID_RESUME_RETRANSMIT_AND_CHANGE_REQ)
+        )
+        {
+
+            gSTWIStatus->msMode = 0;
```
Comment (jiangzhengwenjz) [inline src/librfu_intr.c]: use `AGB_CLK_` macro for this field

File: include/librfu.h [hunk 5]
```diff
@@ -90,8 +91,9 @@
```
Comment (jiangzhengwenjz) [inline include/librfu.h]: remove todo item 1 and 3 in this file

File: include/librfu.h [hunk 6]
```diff
@@ -461,27 +461,15 @@ struct RfuStatic
...
-
 extern struct STWIStatus *gSTWIStatus;
 extern struct RfuLinkStatus *gRfuLinkStatus;
 extern struct RfuStatic *gRfuStatic;
 extern struct RfuFixed *gRfuFixed;
 extern struct RfuSlotStatusNI *gRfuSlotStatusNI[RFU_CHILD_MAX];
 extern struct RfuSlotStatusUNI *gRfuSlotStatusUNI[RFU_CHILD_MAX];
-extern struct RfuSIO32Id gRfuSIO32Id;
 
 // librfu_s32id
```
Comment (jiangzhengwenjz) [inline include/librfu.h]: The comment needs to be updated

File: src/librfu_sio32id.c [hunk 7]
```diff
@@ -58,107 +70,105 @@ static void Sio32IDInit(void)
...
     case 0:
         gRfuSIO32Id.unk0 = 1;
         REG_SIOCNT |= SIO_38400_BPS;
-        REG_IME = r12;
+        REG_IME = 0;
         REG_IE |= INTR_FLAG_SERIAL;
         REG_IME = 1;
-        gRfuSIO32Id.unk1 = 1;
+        gRfuSIO32Id.state = 1;
         *(vu8 *)&REG_SIOCNT |= SIO_ENABLE;
```
Comment (jiangzhengwenjz) [inline src/librfu_sio32id.c]: `*(vu8 *)REG_ADDR_SIOCNT`

File: src/mevent_show_news.c [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+const struct UnkStruct_8467FB8 sBgSpecs[] = {
+    {1, 0, 0, 0, sNews0Gfx, sNews0Map, sNews1Pal},
+    {1, 0, 0, 0, sNews1Gfx, sNews1Map, gCard1Pal},
+    {1, 0, 0, 0, sNews2Gfx, sNews2Map, gCard2Pal},
+    {1, 0, 0, 0, sNews2Gfx, sNews2Map, gCard3Pal},
+    {1, 0, 0, 0, sNews2Gfx, sNews2Map, gCard4Pal},
+    {1, 0, 0, 0, sNews2Gfx, sNews2Map, gCard5Pal},
+    {1, 0, 0, 0, sNews6Gfx, sNews6Map, sNews6Pal},
+    {1, 0, 0, 0, sNews7Gfx, sNews7Map, sNews7Pal}
+};
```
Comment (jiangzhengwenjz) [inline src/mevent_show_news.c]: static

File: src/mevent_show_news.c [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+
+s32 FadeToWonderNewsMenu(void)
+{
+    if (sWork == NULL)
+        return -1;
+
+    switch (sWork->state)
+    {
+    case 0:
+        BeginNormalPaletteFade(0xFFFFFFFF, 0, 0, 16, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_news.c]: `RGB_BLACK`

File: src/mevent_show_news.c [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+        if (UpdatePaletteFade())
+            return 0;
+        ChangeBgY(0, 0, 0);
+        ChangeBgY(1, 0, 0);
+        ChangeBgY(2, 0, 0);
+        ChangeBgY(3, 0, 0);
+        SetGpuReg(REG_OFFSET_WIN0H, 0xF0);
+        SetGpuReg(REG_OFFSET_WIN0V, 0x1A98);
+        SetGpuReg(REG_OFFSET_WININ, 0x1F);
+        SetGpuReg(REG_OFFSET_WINOUT, 0x1B);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_news.c]: constants

File: src/mevent_show_news.c [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+        CopyBgTilemapBufferToVram(0);
+        CopyBgTilemapBufferToVram(2);
+        break;
+    case 6:
+        ShowBg(1);
+        ShowBg(2);
+        ShowBg(3);
+        gPaletteFade.bufferTransferDisabled = FALSE;
+        sWork->menuIndicatorsId = AddScrollIndicatorArrowPair(&sWork->scrollArrowsTemplate, &sWork->scrollOffset);
+        BeginNormalPaletteFade(0xFFFFFFFF, 0, 16, 0, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_news.c]: `RGB_BLACK`

File: src/mevent_show_news.c [hunk 12]
```diff
@@ -0,0 +1,357 @@
...
+}
+
+s32 FadeOutFromWonderNews(bool32 flag)
+{
+    if (sWork == NULL)
+        return -1;
+    switch (sWork->state)
+    {
+    case 0:
+        BeginNormalPaletteFade(0xFFFFFFFF, 0, 0, 16, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_news.c]: `RGB_BLACK`

File: src/mevent_show_news.c [hunk 13]
```diff
@@ -0,0 +1,357 @@
...
+        }
+        break;
+    case 5:
+        PrintMysteryGiftOrEReaderTopMenu(gGiftIsFromEReader, flag);
+        break;
+    case 6:
+        MG_DrawCheckerboardPattern();
+        CopyBgTilemapBufferToVram(0);
+        CopyBgTilemapBufferToVram(3);
+        BeginNormalPaletteFade(0xFFFFFFFF, 0, 16, 0, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_news.c]: `RGB_BLACK`

File: src/wireless_communication_status_screen.c [hunk 14]
```diff
@@ -168,58 +164,58 @@ void sub_814F19C(void)
...
 }
 
-void sub_814F1E4(void)
+static void CB2_InitWirelessCommunicationScreen(void)
 {
     SetGpuReg(REG_OFFSET_DISPCNT, 0);
-    gUnknown_3002040 = AllocZeroed(sizeof(*gUnknown_3002040));
+    sWCSS = AllocZeroed(sizeof(*sWCSS));
     SetVBlankCallback(NULL);
     ResetBgsAndClearDma3BusyFlags(0);
```
Comment (jiangzhengwenjz) [inline src/wireless_communication_status_screen.c]: `FALSE`

File: src/wireless_communication_status_screen.c [hunk 15]
```diff
@@ -228,55 +224,55 @@ void sub_814F32C(void)
...
 
-void sub_814F3A8(void)
+static void PrintHeaderTexts(void)
 {
     s32 i;
     u32 width;
 
     FillWindowPixelBuffer(0, 0);
     FillWindowPixelBuffer(1, 0);
     FillWindowPixelBuffer(2, 0);
```
Comment (jiangzhengwenjz) [inline src/wireless_communication_status_screen.c]: `PIXEL_FILL`

File: include/link_rfu.h [hunk 16]
```diff
@@ -4,53 +4,168 @@
 #include "global.h"
 #include "librfu.h"
 
+//-----------------------------------------------------------------
+// Constant definition
+//-----------------------------------------------------------------
+
```
Comment (jiangzhengwenjz) [inline include/link_rfu.h]: I think we can create link_manager.h to separate SDK code from GF code

File: src/link_rfu_3.c [hunk 17]
```diff
@@ -879,7 +892,7 @@ void RecordMixTrainerNames(void)
         s32 connectedTrainerRecordIndices[5];
         struct TrainerNameRecord *newRecords = AllocZeroed(20 * sizeof(struct TrainerNameRecord));
 
-        // Check if we already have a record saved for connected trainers.
+        // Check if we alsendy have a record saved for connected trainers.
```
Comment (jiangzhengwenjz) [inline src/link_rfu_3.c]: Why is this changed (and the below one)?

File: src/mevent_show_card.c [hunk 18]
```diff
@@ -0,0 +1,454 @@
...
+    /*01DD*/ u8 unk_01DD[7];
+    /*01E4*/ u8 mainMessageLines[4][41];
+    /*0288*/ u8 instructionsLine1[41];
+    /*02B1*/ u8 instructionsLine2[41];
+    /*02DC*/ struct MEventScreenMgr_02DC recordStrings[8];
+    /*045C*/ u8 buffer_045C[0x1000];
+};
+
+static EWRAM_DATA struct MEventScreenMgr * sMEventScreenData = NULL;
+
```
Comment (jiangzhengwenjz) [inline src/mevent_show_card.c]: add `static` qualifier when applicable

File: src/mevent_show_card.c [hunk 19]
```diff
@@ -0,0 +1,454 @@
...
+}
+
+s32 FadeToWonderCardMenu(void)
+{
+    if (sMEventScreenData == NULL)
+        return -1;
+    switch(sMEventScreenData->state)
+    {
+        case 0:
+            BeginNormalPaletteFade(0xFFFFFFFF, 0, 0, 16, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_card.c]: `RGB_BLACK`

File: src/mevent_show_card.c [hunk 20]
```diff
@@ -0,0 +1,454 @@
...
+            break;
+        case 6:
+            LoadMonIconPalettes();
+            break;
+        case 7:
+            ShowBg(1);
+            ShowBg(2);
+            gPaletteFade.bufferTransferDisabled = FALSE;
+            sub_8146060();
+            BeginNormalPaletteFade(0xFFFFFFFF, 0, 16, 0, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_card.c]: `RGB_BLACK`

File: src/mevent_show_card.c [hunk 21]
```diff
@@ -0,0 +1,454 @@
...
+}
+
+s32 FadeOutFromWonderCard(bool32 flag)
+{
+    if (sMEventScreenData == NULL)
+        return -1;
+    switch (sMEventScreenData->state)
+    {
+        case 0:
+            BeginNormalPaletteFade(0xFFFFFFFF, 0, 0, 16, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_card.c]: `RGB_BLACK`

File: src/mevent_show_card.c [hunk 22]
```diff
@@ -0,0 +1,454 @@
...
+        case 4:
+            sub_81461D8();
+            FreeMonIconPalettes();
+            break;
+        case 5:
+            PrintMysteryGiftOrEReaderTopMenu(gGiftIsFromEReader, flag);
+            break;
+        case 6:
+            CopyBgTilemapBufferToVram(0);
+            BeginNormalPaletteFade(0xFFFFFFFF, 0, 16, 0, 0);
```
Comment (jiangzhengwenjz) [inline src/mevent_show_card.c]: `RGB_BLACK`

File: src/link_rfu.c [hunk 23]
```diff
@@ -834,339 +834,338 @@ static void sub_80FDC98(u16 reqCommandId, u16 reqResult)
...
+        lman.nameAcceptTimer.count[lman.child_slot] = 0;
+        rfu_LMAN_occureCallback(LMAN_MSG_CHILD_NAME_SEND_COMPLETED, 0);
     }
 }
 
-static void sub_80FE74C(void)
+static void rfu_LMAN_CHILD_linkRecoveryProcess(void)
 {
-    if (gUnknown_3005E10.unk_06 == 0 && gUnknown_3005E10.unk_0a == 1)
+    if (lman.parent_child == 0 && lman.linkRecovery_start_flag == 1)
```
Comment (jiangzhengwenjz) [inline src/link_rfu.c]: `MODE_CHILD`

File: src/link_rfu.c [hunk 24]
```diff
@@ -2,190 +2,188 @@
...
 
-u8 sub_80FD538(u8 r5, u16 r7, u16 r8, const u16 *r6)
+u8 rfu_LMAN_establishConnection(u8 parent_child, u16 connect_period, u16 name_accept_period, u16 *acceptable_serialNo_list)
 {
     u8 i;
-    const u16 *buffer;
+    u16 *serial_list;
 
-    if (gUnknown_3005E10.unk_04 != 0 && (gUnknown_3005E10.unk_04 != 0x08 || r5 != 1))
+    if (lman.state != LMAN_STATE_READY && (lman.state != LMAN_STATE_WAIT_RECV_CHILD_NAME || parent_child != 1))
```
Comment (jiangzhengwenjz) [inline src/link_rfu.c]: `MODE_PARENT`

File: src/link_rfu.c [hunk 25]
```diff
@@ -2,190 +2,188 @@
...
+        parent_child = 1;
+        connect_period = 0;
     }
     else
     {
-        gUnknown_3005E10.unk_07 = 0;
+        lman.pcswitch_flag = 0;
     }
-    if (r5 != 0)
+    if (parent_child != 0)
```
Comment (jiangzhengwenjz) [inline src/link_rfu.c]: `MODE_CHILD`

File: src/link_rfu.c [hunk 26]
```diff
@@ -196,636 +194,638 @@ u8 sub_80FD610(u16 parentId, u16 unk_1a)
...
-        gUnknown_3005E10.unk_16 = linkLossReason;
-        if (gUnknown_3005E10.unk_09)
+        lman.param[0] = bm_linkLossSlot;
+        lman.param[1] = reason;
+        if (lman.linkRecovery_enable)
         {
-            gUnknown_3005E10.unk_0a = 1;
-            if (gUnknown_3005E10.unk_06 == 0 && linkLossReason == REASON_DISCONNECTED)
+            lman.linkRecovery_start_flag = 1;
+            if (lman.parent_child == 0 && reason == REASON_DISCONNECTED)
```
Comment (jiangzhengwenjz) [inline src/link_rfu.c]: `MODE_CHILD`

File: src/link_rfu.c [hunk 27]
```diff
@@ -834,339 +834,338 @@ static void sub_80FDC98(u16 reqCommandId, u16 reqResult)
...
                 {
-                    if ((gUnknown_3005E10.unk_14 >> i) & 1)
+                    if ((lman.param[0] >> i) & 1)
                     {
-                        gUnknown_3005E10.unk_34[i] = 0;
+                        lman.linkRecoveryTimer.count[i] = 0;
                     }
                 }
-                if (gUnknown_3005E10.unk_06 == 0)
+                if (lman.parent_child == 0)
```
Comment (jiangzhengwenjz) [inline src/link_rfu.c]: `MODE_CHILD`

File: src/link_rfu_2.c [hunk 28]
```diff
@@ -51,28 +52,28 @@ static void sub_80FA9D0(u16 a0);
 static void sub_80FAA58(void * a0);
 static void sub_80FAA94(u8 taskId);
 static void sub_80FACF0(u8 taskId);
-static void sub_80FB0E8(u32 a0);
+static void GetLinkmanErrorParams(u32 msg);
 static void sub_80FB564(s32 a0);
 static void sub_80FBB74(void);
-static u8 sub_80FBC70(const u8 *a0, u16 a1);
-static void sub_80FBCF8(u32 a0);
+static u8 GetPartnerIndexByNameAndTrainerID(const u8 *trainerName, u16 trainerId);
```
Comment (jiangzhengwenjz) [inline src/link_rfu_2.c]: ID->Id

File: src/librfu_rfu.c [hunk 29]
```diff
@@ -91,46 +116,46 @@ static const char lib_ver[] = "RFU_V1024";
...
+        if (buffByteSize < buffByteSizeMax)
             return ERR_RFU_API_BUFF_SIZE;
     }
-    gRfuLinkStatus = &APIBuffer->linkStatus;
-    gRfuStatic = &APIBuffer->static_;
-    gRfuFixed = &APIBuffer->fixed;
-    gRfuSlotStatusNI[0] = &APIBuffer->NI[0];
-    gRfuSlotStatusUNI[0] = &APIBuffer->UNI[0];
+    gRfuLinkStatus = (void *)APIBuffer + 0;
+    gRfuStatic = (void *)APIBuffer + 0xb4;
```
Comment (jiangzhengwenjz) [inline src/librfu_rfu.c]: Can you update the constants with some sizeof or comment

File: src/librfu_rfu.c [hunk 30]
```diff
@@ -1784,434 +1819,433 @@ void rfu_REQ_recvData(void)
...
-        r5->newDataFlag = 1;
-        r5->state = 0;
+        UNI_recv->state = SLOT_STATE_RECEIVING;
+        size = UNI_recv->dataSize = llsf_NI->unk6;
+        dest = gRfuSlotStatusUNI[bm_slot_id]->recvBuffer;
+        gRfuFixed->fastCopyPtr(&src, &dest, size);
+        UNI_recv->newDataFlag = 1;
+        UNI_recv->state = 0;
     }
 _081E2F0E:
```
Comment (jiangzhengwenjz) [inline src/librfu_rfu.c]: The label can now be documented

File: include/librfu.h [hunk 31]
```diff
@@ -90,8 +88,9 @@
 
 #define RFU_MBOOT_DOWNLOADER_SERIAL_NO  0x0000             // The game serial number of the multi-boot downloader (programs that boot without a Game Pak)
 
-#define RFU_API_BUFF_SIZE_RAM           0x0e8c             // Necessary size for buffer specified by rfu_initializeAPI (fast communication version that operates the library SIO interrupt routines in RAM)
-#define RFU_API_BUFF_SIZE_ROM           0x052c             // Necessary size for buffer specified by rfu_initializeAPI (fast communication version that operates the library SIO interrupt routines in ROM)
+// Sizes are 0x28 larger in v1028
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/librfu.h]: not a significant enough detail to correct

File: src/link_rfu_3.c (hunk 17 above)
Comment (PikalaxALT) [reply src/link_rfu_3.c]: `s/read/send/g` 🤦‍♂

File: src/link_rfu_2.c (hunk 28 above)
Comment (PikalaxALT) [reply src/link_rfu_2.c]: IMO "ID" should be in all caps because it's like that in-game

File: src/librfu_sio32id.c (hunk 7 above)
Comment (jiangzhengwenjz) [reply src/librfu_sio32id.c]: You missed this one

File: src/union_room.c [hunk 32]
```diff
@@ -578,47 +664,56 @@ static const u8 gUnknown_84570C8[] = {0x01, 0x02, 0x03, 0x04, 0x09, 0x0A, 0x0B,
     _maxWidth - strWidth;                                          \
 })
 
-static void sub_811586C(u8 windowId, u8 arg1, u8 stringId)
+// arg1 is a 2-nybble code
```
Comment (GriffinRichards) [inline src/union_room.c]: capacityCode

File: include/constants/union_room.h [hunk 33]
```diff
@@ -1,6 +1,48 @@
...
+// Player response
+#define ACTIVITY_ACCEPT    17
+#define ACTIVITY_DECLINE   18
+
+#define ACTIVITY_NPCTALK   19
+#define ACTIVITY_PLYRTALK  20
+
+// Duplicate IDs?
+#define ACTIVITY_WCARD2    21
+#define ACTIVITY_WNEWS2    22
```
Comment (GriffinRichards) [inline include/constants/union_room.h]: Not super important but it seems like the other ones are the duplicates, and might be clearer if they were the 2. Are the first ones used anywhere other than `sUnionRoomActivityStringPtrs`?

File: src/AgbRfu_LinkManager.c [hunk 34]
```diff
@@ -0,0 +1,2508 @@
...
+													 (Note: Use this function or rfu_LMAN_setMSC callback for the MSC callback settings when using the Link Manager. Do not use rfu_REQ_setMSCCallback.
+															Furthermore, REQ callbacks cannot be used.)
+[Argument return value]	None
+
+[Function return value] u8 errorCode				:0 -> Normal end
+ 													 Issued when LMAN_ERROR_ILLEGAL_PARAMETER -> LMAN_callback_p is not specified.
+------------------------------------------------------------------*/
+u8 rfu_LMAN_initializeManager(void (*LMAN_callback_p)(u8, u8), void (*MSC_callback_p)(u16))
+{
+    if ((u32) LMAN_callback_p == 0)
```
Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `if ((uintptr_t)LMAN_callback_p == NULL)`

File: src/AgbRfu_LinkManager.c [hunk 35]
```diff
@@ -0,0 +1,2508 @@
...
+    {
+        return LMAN_ERROR_ILLEGAL_PARAMETER;
+    }
+
+    CPU_FILL(0x0000, &lman, sizeof(LINK_MANAGER), 16);
```
Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `CpuFill16`

File: src/AgbRfu_LinkManager.c [hunk 36]
```diff
@@ -0,0 +1,2508 @@
...
+[Argument] 			None
+
+[Argument return value]	None
+
+[Function return value]
+
+------------------------------------------------------------------*/
+static void rfu_LMAN_endManager(void)
+{
+    CPU_FILL(0x0000, &lman, sizeof(LINK_MANAGER) - 8, 16);
```
Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `CpuFill16`

File: src/AgbRfu_LinkManager.c [hunk 37]
```diff
@@ -0,0 +1,2508 @@
...
+				LMAN_MSG_SEARCH_CHILD_PERIOD_EXPIRED		0								-								-			Generated when the SearchChild run period expires.
+				LMAN_MSG_END_WAIT_CHILD_NAME				0								-								-			Generated when reception of game ID information from all children and connection decisions have completed after the SearchChild run period expires.
+				(Generated by child device)
+				LMAN_MSG_PARENT_FOUND						1								Bit indicating 					-			Generated when a valid (game serial numbers match) parent device is found while SearchParent is running.
+																							the index number of gRfuLinkStatus->partner[0-3] where information about the parent with a matching game serial number is stored.
+				LMAN_MSG_SEARCH_PARENT_PERIOD_EXPIRED		0								-								-			Generated when SearchParentperiod ends.
+				(Generated when both parent and child devices issue an error)
+				LMAN_MSG_REQ_API_ERROR						2							REQ_commandID					REQ_result
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+u8
```
Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: :skull: why does it exchange to GNU style

File: src/AgbRfu_LinkManager.c [hunk 38]
```diff
@@ -0,0 +1,2508 @@
...
+				LMAN_MSG_LINK_RECOVERY_FAILED_AND_DISCONNECTED		1			Bit indicating slot 		-			Generated when link recovery fails and that slot disconnects. (Generated only when link recovery is ON.)
+																				where link recovery failed and that was disconnected
+				(Generated when an error occurs)
+				LMAN_MSG_REQ_API_ERROR								2			REQ_commandID				REQ_result
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+void rfu_LMAN_PARENT_stopWaitLinkRecoveryAndDisconnect(u8 bm_targetSlot)
+{
+    u8 i;
+
+    if ((bm_targetSlot & lman.linkRecoveryTimer.active) == 0)
```
Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `if (!(bm_targetSlot & lman.linkRecoveryTimer.active))`

File: src/AgbRfu_LinkManager.c [hunk 39]
```diff
@@ -0,0 +1,2508 @@
...
+
+[Argument return value]	None
+
+[Function return value] None
+------------------------------------------------------------------*/
+void rfu_LMAN_manager_entity(u32 rand)
+{
+    u8 msg;
+
+    if (((u32) lman.LMAN_callback == 0x00000000) && (lman.state))
```
Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `if (lman.LMAN_callback == NULL && lman.state)`

File: src/AgbRfu_LinkManager.c [hunk 40]
```diff
@@ -0,0 +1,2508 @@
...
+            }
+            break;
+
+        case ID_CP_END_REQ:
+            if (REQ_result == 0)
+            {
+                if (rfu_getConnectParentStatus(&status, &lman.child_slot) == 0)
+                {    // Final check for connection processing.
+                    if (status == CP_STATUS_DONE)
+                    {
```
Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: space

Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `if (lman.MSC_callback != NULL)`

Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `if (!(flags & RN_ACCEPT))`

Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: `!= NULL`

Comment (jiangzhengwenjz) [inline src/AgbRfu_LinkManager.c]: redundant newline

File: src/trade.c [hunk 41]
```diff
@@ -2746,7 +2746,7 @@ int GetUnionRoomTradeMessageId(struct GFtgtGnameSub a0, struct GFtgtGnameSub a1,
     u8 r4 = a0.isChampion;
     u8 r10 = a1.hasNationalDex;
     u8 r0 = a1.isChampion;
-    u8 r1 = a1.unk_01_2;
+    u8 r1 = a1.version;
     u8 r2;
```
Comment (jiangzhengwenjz) [inline src/trade.c]: These can all be documented. `r2` is `bool8 isNotFRLG;`

File: src/link.c [hunk 42]
```diff
@@ -1477,12 +1477,12 @@ void CB2_LinkError(void)
         {
             gWirelessCommType = 3;
         }
-        sub_80F85F8();
+        ResetLinkRfuGFLayer();
     }
     SetVBlankCallback(sub_800978C);
     ResetBgsAndClearDma3BusyFlags(0);
```
Comment (jiangzhengwenjz) [inline src/link.c]: `FALSE`.

File: src/link.c [hunk 43]
```diff
@@ -82,11 +82,11 @@ u32 gFiller_3003EC0;
...
+bool8 gLinkAllAcked5FFF;
 bool8 gUnknown_3003F28;
-bool8 gUnknown_3003F2C[MAX_LINK_PLAYERS];
-bool8 gUnknown_3003F30[MAX_LINK_PLAYERS];
-u16 gUnknown_3003F34;
+bool8 gLinkCommand2FFEAck[MAX_LINK_PLAYERS];
+bool8 gLinkCommand5FFFAck[MAX_LINK_PLAYERS];
+u16 gLinkCmd5FFFparam;
 u8 gSuppressLinkErrorMessage;
 bool8 gWirelessCommType;
```
Comment (jiangzhengwenjz) [inline src/link.c]: This is `u8`

File: src/link.c [hunk 44]
```diff
@@ -82,11 +82,11 @@ u32 gFiller_3003EC0;
...
 bool8 gSavedLinkPlayerCount;
```
Comment (jiangzhengwenjz) [inline src/link.c]: `u8`

File: src/link_rfu_2.c [hunk 45]
```diff
@@ -433,38 +437,38 @@ static void sub_80F8B34(u8 taskId)
...
 {
-    sub_80FD538(1, 0, 240, gUnknown_843EC8C);
+    rfu_LMAN_establishConnection(MODE_PARENT, 0, 240, (u16*)sAcceptedSerialNos);
 }
 
-void sub_80F8D14(void)
+void LinkRfu_StopManagerBeforeEnteringChat(void)
 {
-    sub_80FD760(FALSE);
+    rfu_LMAN_stopManager(FALSE);
```
Comment (jiangzhengwenjz) [inline src/link_rfu_2.c]: `When 0 is specified for the u8 forced_stop_and_rfu_reset argument here, the user can continue with a new connection operation` so `FALSE` doesn't really make sense here

File: data/event_scripts.s [hunk 46]
```diff
@@ -1170,9 +1171,9 @@ EventScript_ResetEliteFour:: @ 81A6551
 	.include "data/scripts/pc.inc"
 
 @ DoEasyChatScreen?
```
Comment (GriffinRichards) [inline data/event_scripts.s]: Remove comment

File: src/trade.c [hunk 47]
```diff
@@ -2740,30 +2740,30 @@ static bool32 IsDeoxysOrMewUntradable(u16 species, bool8 isObedientBitSet)
...
-    u8 r10 = a1.hasNationalDex;
-    u8 r0 = a1.isChampion;
-    u8 r1 = a1.unk_01_2;
-    u8 r2;
+    u8 playerHasNationalDex = playerSub.hasNationalDex;
+    u8 playerIsChampion = playerSub.isChampion;
+    u8 partnerHasNationalDex = partnerSub.hasNationalDex;
+    u8 partnerIsChampion = partnerSub.isChampion;
+    u8 partnerVersion = partnerSub.version;
+    u8 isNotFRLG;
```
Comment (GriffinRichards) [inline src/trade.c]: `bool8`

File: include/union_room.h [hunk 48]
```diff
@@ -4,28 +4,34 @@
...
 #include "link_rfu.h"
 
-struct UnkStruct_Shared
+// Return value of IsRequestedTypeAndSpeciesInPlayerParty
+#define UR_TRADE_MATCH  0
+#define UR_TRADE_NOTYPE 1
+#define UR_TRADE_NOEGG  2
+
+#define UROOM_MAX_GROUP_COUNT 8
+#define UROOM_MAX_PARTY_SIZE  5
```
Comment (GriffinRichards) [inline include/union_room.h]: Is this used? I associate party size with pokemon party size, and I havent heard of a union room pokemon party limit

File: src/trade.c [hunk 49]
```diff
@@ -2740,30 +2740,30 @@ static bool32 IsDeoxysOrMewUntradable(u16 species, bool8 isObedientBitSet)
...
-    u8 r9 = a0.hasNationalDex;
-    u8 r4 = a0.isChampion;
-    u8 r10 = a1.hasNationalDex;
-    u8 r0 = a1.isChampion;
-    u8 r1 = a1.unk_01_2;
-    u8 r2;
+    u8 playerHasNationalDex = playerSub.hasNationalDex;
+    u8 playerIsChampion = playerSub.isChampion;
+    u8 partnerHasNationalDex = partnerSub.hasNationalDex;
+    u8 partnerIsChampion = partnerSub.isChampion;
```
Comment (jiangzhengwenjz) [inline src/trade.c]: `bool8`

File: src/link.c [hunk 50]
```diff
@@ -1683,23 +1683,23 @@ bool8 HandleLinkConnection(void)
     return FALSE;
 }
 
-void sub_800B1F4(void)
+void SetWirelessCommType1(void)
 {
     if (gReceivedRemoteLinkPlayers == 0)
```
Comment (jiangzhengwenjz) [inline src/link.c]: `!gReceivedRemoteLinkPlayers`

File: src/link.c [hunk 51]
```diff
@@ -1683,23 +1683,23 @@ bool8 HandleLinkConnection(void)
...
     {
         gWirelessCommType = 1;
     }
 }
 
-static void sub_800B210(void)
+static void SetWirelessCommType0(void)
 {
     if (gReceivedRemoteLinkPlayers == 0)
```
Comment (jiangzhengwenjz) [inline src/link.c]: `!gReceivedRemoteLinkPlayers`

File: src/link.c [hunk 52]
```diff
@@ -1683,23 +1683,23 @@ bool8 HandleLinkConnection(void)
...
     {
         gWirelessCommType = 0;
     }
 }
 
-void sub_800B22C(void)
+void SetWirelessCommType0_UnusedCopy(void)
 {
     if (gReceivedRemoteLinkPlayers == 0)
```
Comment (jiangzhengwenjz) [inline src/link.c]: `!gReceivedRemoteLinkPlayers`
maybe do a pattern search to fix all use cases of the variable
